### PR TITLE
7.2/59090 symfony security

### PR DESCRIPTION
**Pull request by rnuernberg**

### CODE REVIEW

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPalExpress.class.php [hunk 1]
```diff
@@ -115,7 +115,7 @@ public function PostProcessExternalPaymentHandlerHook()
      *
      * @return void
      */
-    protected function GetUserDataFromPayPalData(&$aBilling, &$aShipping)
+    protected function GetUserDataFromPayPalData($aBilling, $aShipping)
```
Comment (bestform) [inline src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPalExpress.class.php:118]: This needs to stay

File: src/ShopBundle/objects/db/TShopVariantDisplayHandler/TShopVariantDisplayHandler.class.php [hunk 2]
```diff
@@ -199,7 +199,7 @@ protected function GetAdditionalViewVariables(&$oArticle, $sViewName, $sViewType
      *
      * @return void
      */
-    protected function AddClearCacheTriggers(&$aClearTriggers, &$oArticle, $sViewName, $sViewType)
+    protected function AddClearCacheTriggers($aClearTriggers, $oArticle, $sViewName, $sViewType)
```
Comment (bestform) [inline src/ShopBundle/objects/db/TShopVariantDisplayHandler/TShopVariantDisplayHandler.class.php:202]: This needs to stay

File: src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariant.class.php [hunk 3]
```diff
@@ -68,7 +68,7 @@ protected function GetVariantType($aRow)
      *
      * @return void
      */
-    protected function OrderOptions(&$aOptions)
+    protected function OrderOptions($aOptions)
```
Comment (bestform) [inline src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariant.class.php:71]: This needs stay

File: src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariantColor.class.php [hunk 4]
```diff
@@ -18,7 +18,7 @@ class TPkgShopListfilterItemVariantColor extends TPkgShopListfilterItemVariant
      * @param array<string, mixed> $aOptions
      * @return void
      */
-    protected function OrderOptions(&$aOptions)
+    protected function OrderOptions($aOptions)
```
Comment (bestform) [inline src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariantColor.class.php:21]: please stay another day

File: src/ShopListFilterBundle/objects/db/TPkgShopListfilterItem.class.php [hunk 5]
```diff
@@ -320,7 +320,7 @@ public function IsActive()
      *
      * @return void
      */
-    protected function OrderOptions(&$aOptions)
+    protected function OrderOptions($aOptions)
```
Comment (bestform) [inline src/ShopListFilterBundle/objects/db/TPkgShopListfilterItem.class.php:323]: Stay

**Review by bestform — CHANGES_REQUESTED** [review]

File: src/ImageHotspotBundle/objects/db/TPkgImageHotspotItem.class.php [hunk 6]
```diff
@@ -64,7 +64,7 @@ protected function GetAdditionalViewVariables($sViewName, $sViewType)
      *
      * @return void
      */
-    protected function AddClearCacheTriggers(&$aClearTriggers, $sViewName, $sViewType)
+    protected function AddClearCacheTriggers($aClearTriggers, $sViewName, $sViewType)
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ImageHotspotBundle/objects/db/TPkgImageHotspotItem.class.php:67]: item caching is no longer used - method removed

File: src/ShopBundle/objects/ArticleList/Module.php [hunk 7]
```diff
@@ -209,7 +209,7 @@ private function initializeListState()
      *
      * @return void
      */
-    private function makePageSizeValid(array &$stateData)
+    private function makePageSizeValid(array $stateData)
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ShopBundle/objects/ArticleList/Module.php:212]: private method - changed to return data

File: src/ShopBundle/objects/TShopBasket/TShopBasketArticleCoreList.class.php [hunk 8]
```diff
@@ -317,7 +317,7 @@ public function UpdateItemAmount(TShopBasketArticle &$oItem)
      *
      * @return TShopBasketArticle|false
      */
-    public function &next()
+    public function next(): TShopBasketArticle|false
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ShopBundle/objects/TShopBasket/TShopBasketArticleCoreList.class.php:320]: sad. changed... also affects the auto classes generated by the core - and any usages of the classes

File: src/ShopBundle/objects/TShopBasket/TShopBasketArticleDiscountCoreList.class.php [hunk 9]
```diff
@@ -19,7 +19,7 @@ class TShopBasketArticleDiscountCoreList extends TIterator
      *
      * @return TdbShopDiscount|false
      */
-    public function &next()
+    public function next(): TdbShopDiscount|false
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ShopBundle/objects/TShopBasket/TShopBasketArticleDiscountCoreList.class.php:22]: done

File: src/ShopBundle/objects/TShopBasket/TShopBasketArticleDiscountCoreList.class.php [hunk 10]
```diff
@@ -39,7 +39,7 @@ public function &Previous()
      *
      * @return TdbShopDiscount
      */
-    public function &current()
+    public function current():TdbShopDiscount|false
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ShopBundle/objects/TShopBasket/TShopBasketArticleDiscountCoreList.class.php:42]: done

File: src/ShopBundle/objects/TShopBasket/TShopBasketDiscountCoreList.class.php [hunk 11]
```diff
@@ -14,7 +14,7 @@ class TShopBasketDiscountCoreList extends TIterator
     /**
      * @return TdbShopDiscount|false
      */
-    public function &next()
+    public function next(): TdbShopDiscount|false
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ShopBundle/objects/TShopBasket/TShopBasketDiscountCoreList.class.php:17]: done

File: src/ShopBundle/objects/TShopBasket/TShopBasketVoucherCoreList.class.php [hunk 12]
```diff
@@ -53,7 +53,7 @@ protected function GetUniqueItemKey()
      *
      * @return TdbShopVoucher|false
      */
-    public function &next()
+    public function next(): TdbShopVoucher|false
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ShopBundle/objects/TShopBasket/TShopBasketVoucherCoreList.class.php:56]: done

File: src/ShopBundle/objects/db/TShopArticle.class.php [hunk 13]
```diff
@@ -1769,7 +1769,7 @@ public function GetBreadcrumbName()
      *
      * @return array SEO pattern replace values
      */
-    public function GetSeoPattern(&$sPaternIn)
+    public function GetSeoPattern($sPaternIn)
```
[the remark this answers is not in the document]
Comment (rnuernberg) [reply src/ShopBundle/objects/db/TShopArticle.class.php:1772]: true

File: src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPalExpress.class.php (hunk 1 above)
Comment (rnuernberg) [reply src/ShopBundle/objects/db/TShopPaymentHandler/TShopPaymentHandlerPayPalExpress.class.php:118]: reverted

File: src/ShopBundle/objects/db/TShopVariantDisplayHandler/TShopVariantDisplayHandler.class.php (hunk 2 above)
Comment (rnuernberg) [reply src/ShopBundle/objects/db/TShopVariantDisplayHandler/TShopVariantDisplayHandler.class.php:202]: method is no longer needed - removed

File: src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariant.class.php (hunk 3 above)
Comment (rnuernberg) [reply src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariant.class.php:71]: reverted

File: src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariantColor.class.php (hunk 4 above)
Comment (rnuernberg) [reply src/ShopListFilterBundle/objects/db/ListfilterItems/TPkgShopListfilterItemVariantColor.class.php:21]: reverted

File: src/ShopListFilterBundle/objects/db/TPkgShopListfilterItem.class.php (hunk 5 above)
Comment (rnuernberg) [reply src/ShopListFilterBundle/objects/db/TPkgShopListfilterItem.class.php:323]: reverted

**Comment by bestform** [comment]
I resolved the conflicts to the best of my knowledge. Could you please have a quick look at my commit to confirm that I didn't do anything stupid?

**Comment by rnuernberg** [comment]
> I resolved the conflicts to the best of my knowledge. Could you please have a quick look at my commit to confirm that I didn't do anything stupid?

I have no idea where to find your commit.

**Comment by rnuernberg** [comment]
> It is the last merge commit: [5978322](https://github.com/chameleon-system/chameleon-shop/commit/59783222cedac62e6f294e447df40b4a7fdffe11)

looks good

**Pull request merged**